### PR TITLE
ENG-6404 - central config docker templates

### PR DESCRIPTION
**Pull request by simhadriraju-digitalai**
## Definition of Done

**General**
 - [ ] Branch is up-to-date with master
 - [ ] Code is reviewed and tested by someone in the Kube team
 - [ ] If there are user facing changes (new env variables for example) update the docs and make sure to notify the docs team to update official documentation

### CODE REVIEW

File: templates/dockerfiles/central-configuration/install.j2 [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /tmp/modify-wrapper-linux-conf.gawk ${APP_HOME}/default-conf/{{ wrapper_conf }} > /tmp/{{ wrapper_conf}} && \
+    mv /tmp/{{ wrapper_conf }} ${APP_HOME}/default-conf/{{ wrapper_conf }} && \
+    rm /tmp/modify-wrapper-linux-conf.gawk
+
+
+# Set permissions
+RUN chgrp -R 0 ${APP_ROOT} && \
+    chmod -R g=u ${APP_ROOT} && \
+    chmod g+x ${APP_HOME}/bin/*.sh && \
+    chmod a+rwx ${APP_HOME}/centralConfiguration
```
Comment (muthuraj-mariappan) [inline templates/dockerfiles/central-configuration/install.j2:46]: do we need to set permission for /conf directory as well?

File: templates/resources/central-configuration/central-conf/logback.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            <pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} [%thread] {%mdc} %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="info">
+        <appender-ref ref="STDOUT" />
+        <appender-ref ref="FILE" />
+    </root>
+
+    <logger name="com.xebialabs" level="info" />
```
Comment (muthuraj-mariappan) [inline templates/resources/central-configuration/central-conf/logback.xml:38]: can we keep only the relevant packages and remove the rest

File: templates/resources/common/central-conf/boot.conf.template.j2 [hunk 3]
```diff
@@ -9,3 +9,4 @@ ssl.mutual=false
 ssl=false
 threads.max=150
 threads.min=30
+xl.spring.cloud.encrypt.key=aaL`;*~`<l-Xv.6
```
Comment (muthuraj-mariappan) [inline templates/resources/common/central-conf/boot.conf.template.j2:12]: there is another boot.conf.template.j2 specific for central-configuration. Do we need this change here?

Comment (simhadriraju-digitalai) [reply templates/resources/common/central-conf/boot.conf.template.j2:12]: This change is not required, forgot to cleanup code, Thanks

File: templates/dockerfiles/central-configuration/install.j2 (hunk 1 above)
Comment (simhadriraju-digitalai) [reply templates/dockerfiles/central-configuration/install.j2:46]: conf folder is created with proper permissions set

File: templates/resources/central-configuration/central-conf/logback.xml (hunk 2 above)
Comment (simhadriraju-digitalai) [reply templates/resources/central-configuration/central-conf/logback.xml]: Removed irrelevant  loggers

File: templates/resources/central-configuration/help.md.j2 [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+APP_ROOT=/opt/xebialabs
+    This environment variable is used to determine the directory where all application specific code will be copied. Normally there is no need to change this variable.
+
+APP_HOME=/${APP_ROOT}/{{ product }}-server
+    This environment variable is used as the landing point for all the application data. Ensure that this is a subdirectory of the ${APP_ROOT} variable. Normally there is no need to change this variable.
+
+# VOLUMES
+The following locations can be mounted as volumes to provide either configuration data, and/or persistent storage for application data.
+
```
Comment (muthuraj-mariappan) [inline templates/resources/central-configuration/help.md.j2:22]: we need to add centralConfiguration volume also

Comment (simhadriraju-digitalai) [reply templates/resources/central-configuration/help.md.j2:22]: corrected the docs

File: templates/resources/central-configuration/help.md.j2 [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+**\${APP_HOME}/conf** - This directory contains the configuration files for Central Configuration. If mounted it is possible to configure (and inject) non-default configuration settings.
+
+**\${APP_HOME}/centralConfiguration** - This directory contains the configuration yaml files for Central Configuration. If mounted it is possible to configure (and inject)  configuration yaml files.
+
+- The `deployit.conf` file is generated by the container startup script based on env variables set, you need to provide values for below in that file
```
Comment (acierto) [inline templates/resources/central-configuration/help.md.j2:27]: How `deployit.conf` related to standalone central configuration?

**Comment by simhadriraju-digitalai** [comment]
Manually triggered build passed, it is good to merge
https://jenkins-ng.xebialabs.com/jenkinsng/blue/organizations/jenkins/Official%20Docker%20Images%2FXLProducts/detail/PR-87/30/pipeline

**Review by muthuraj-mariappan — APPROVED** [review]

**Pull request merged**